src/runtime_src/core/common/api/xrt_kernel.cpp

```diff
@@ -699,6 +699,19 @@ class kernel_command : public xrt_core::command
     XRT_DEBUGF("kernel_command::kernel_command(%d)\n", m_uid);
   }
 
+  explicit
+  kernel_command(std::shared_ptr<device_type> dev, xrt_core::hw_queue hwqueue, xrt::hw_context hwctx)
+      : m_device(std::move(dev))
+      , m_hwqueue(std::move(hwqueue))
+      , m_hwctx(std::move(hwctx))
+      , m_execbuf(m_device->create_exec_buf<ert_start_kernel_cmd>())
+      , m_done(true)
+  {
+      static unsigned int count = 0;
+      m_uid = count++;
+      XRT_DEBUGF("kernel_command::kernel_command(%d)\n", m_uid);
+  }
```

Comment:
using original constructor now, where I added new argument as hw_ctx with default initialization. So, it will not create problem when 2 args passed to this constructor( this is used in copy_bo_with_kdma()).